packages/cubejs-server-core/src/core/CompilerApi.js

```diff
@@ -29,6 +30,11 @@ export class CompilerApi {
     this.sqlCache = options.sqlCache;
     this.standalone = options.standalone;
     this.nativeInstance = this.createNativeInstance();
+    this.compiledScriptCache = new LRUCache({
+      max: options.compilerCacheSize || 250,
```

Comment:
I didn't test, but I assume that `vm.Script` takes ~100kb for each "medium" complex model (file). Then 250 is ~25 MB. Is it correct?  But the biggest problem is that `lru-cache` doesn't proactively remove expired items, which can leave a lot of time.

For `CompilerApi` instance cache we are using:

```
    // proactively free up old cache values occasionally
    if (this.options.maxCompilerCacheKeepAlive) {
      this.maxCompilerCacheKeep = setInterval(
        () => this.compilerCache.prune(),
        this.options.maxCompilerCacheKeepAlive
      );
    }
```

I suggest do the same, but we need to enable `ttl` here by default.

Reply:
It uses the same options passed from the core/server: 
```
      max: this.options.compilerCacheSize || 250,
      ttl: this.options.maxCompilerCacheKeepAlive,
      updateAgeOnGet: this.options.updateCompilerCacheKeepAlive
```
So we should be fine here.

I can add the `setInterval` for pro-active cleanups. The question is when to clear it....